packages/next/src/build/webpack-config.ts

```diff
@@ -445,6 +445,21 @@ export default async function getBaseWebpackConfig(
     loggedIgnoredCompilerOptions = true
   }
 
+  const excludeCache: Record<string, boolean> = {}
+  function exclude(excludePath: string): boolean {
+    if (excludeCache[excludePath] !== undefined) {
+      return excludeCache[excludePath]
+    }
```

Comment:
This is probably optimized by V8's JIT, but just to be sure...

```suggestion
    const cached = excludeCache[excludePath]
    if (cached !== undefined) {
      return cached
    }
```